gkr_iop/examples/multi_layer_logup.rs

```diff
@@ -0,0 +1,283 @@
+use std::{marker::PhantomData, mem, sync::Arc};
+
+use ff_ext::ExtensionField;
+use gkr_iop::{
+    ProtocolBuilder, ProtocolWitnessGenerator,
+    chip::Chip,
+    evaluation::{EvalExpression, PointAndEval},
+    gkr::{
+        GKRCircuitWitness, GKRProverOutput,
+        layer::{Layer, LayerType, LayerWitness},
+        mock::MockProver,
+    },
+};
+use goldilocks::GoldilocksExt2;
+use itertools::{Itertools, izip};
+use rand::{Rng, rngs::OsRng};
+use subprotocols::expression::{Constant, Expression, VectorType};
+use transcript::{BasicTranscript, Transcript};
+
+type E = GoldilocksExt2;
+
+#[derive(Clone, Debug, Default)]
+struct TowerParams {
+    height: usize,
+}
+
+#[derive(Clone, Debug, Default)]
+struct TowerChipLayout<E> {
+    params: TowerParams,
+
+    // Commit poly indices.
+    committed_table: usize,
```

Comment:
we could just call `committed_table_id`, `committed_count_id`, more self-explanation